byterun/signals.c

```diff
@@ -76,7 +76,7 @@ void caml_record_signal(int signal_number)
 static caml_root caml_signal_handlers;
 
 void caml_init_signal_handling() {
-  caml_signal_handlers = caml_create_root(caml_alloc(NSIG, 0));
```

Comment:
`caml_alloc` may trigger a GC, but at this point GC is not fully initialized. In particular, the frame tables haven't been inited. By allocating in the shared heap, we ensure that GC cannot be triggered due to a failed allocation. This case was triggered on aarch64 but the fix applies to both architectures.